java/src/main/java/com/cybersource/ws/client/HttpClientConnection.java

```diff
@@ -160,40 +162,48 @@ InputStream getResponseErrorStream()
     }
 
     /**
-     * Helps to set the timeout for HTTP request call.
-     * cybs.properties can be used to configure the timeout details.
-     * @param httpClient
-     * @param timeoutInMs
+     * @param mc
      */
-    private void setTimeout(HttpClient httpClient, int timeoutInMs) {
-        HttpConnectionManagerParams params
-                = httpClient.getHttpConnectionManager().getParams();
-        params.setConnectionTimeout(timeoutInMs);
-        params.setSoTimeout(timeoutInMs);
+    protected void initHttpClient(MerchantConfig mc, int timeoutInMs) {
+        RequestConfig.Builder requestConfigBuilder = RequestConfig.custom()
+                .setSocketTimeout(timeoutInMs)
+                .setConnectTimeout(timeoutInMs);
+
+        HttpClientBuilder httpClientBuilder = HttpClients.custom();
+
+        if(mc.isAllowRetry()){
+            httpClientBuilder.setRetryHandler(new MyRetryHandler());
+        }
+
+        setProxy(httpClientBuilder, requestConfigBuilder, mc);
+
+        httpClient = httpClientBuilder
+                .setDefaultRequestConfig(requestConfigBuilder.build())
+                .build();
     }
 
     /**
-     * This is useful in Client environment where Firewall is set to prevent accessing the external services.
-     * Proxy settings are required in such scenarios. 
-     * @param httpClient
+     * Set proxy by using proxy credentials to create httpclient
+     *
+     * @param httpClientBuilder
+     * @param requestConfigBuilder
+     * @param merchantConfig
      */
-    private void setProxy(HttpClient httpClient) {
-        if (mc.getProxyHost() != null) {
-            httpClient.getHostConfiguration().setProxy(
-                    mc.getProxyHost(), mc.getProxyPort());
+    private void setProxy(HttpClientBuilder httpClientBuilder, RequestConfig.Builder requestConfigBuilder, MerchantConfig merchantConfig) {
```

Comment:
it seems to be common method as in poolingHttpClient. See if we can put this in some common place or in ConnectionHelper class. And refer this method from both the places.